mailer/engine.py

```diff
@@ -154,19 +154,35 @@ def send_all():
                     logging.warning("message discarded due to failure in converting from DB. Added on '%s' with priority '%s'" % (message.when_added, message.priority))  # noqa
                 message.delete()
 
-            except (socket_error, smtplib.SMTPSenderRefused,
-                    smtplib.SMTPRecipientsRefused,
-                    smtplib.SMTPDataError,
-                    smtplib.SMTPAuthenticationError) as err:
+            except (smtplib.SMTPConnectError,
+                    smtplib.SMTPServerDisconnected) as err:
+                # Connection-related problem -- try again soon:
+                logging.info("message left queued due to transient failure: %s" % err)
+                MessageLog.objects.log(message, RESULT_FAILURE, log_message=str(err))
+                break  # connectivity issue; resume batch next time
+
+            except (socket_error,  # includes smtplib.SMTPException in python3.4+
+                    # TODO: consider handling socket_error/OSError/IOError as "retry soon"???
+                    smtplib.SMTPException) as err:
+                # Destination/recipient-related problem -- try again later:
                 message.defer()
                 logging.info("message deferred due to failure: %s" % err)
                 MessageLog.objects.log(message, RESULT_FAILURE, log_message=str(err))
                 deferred += 1
                 # Get new connection, it case the connection itself has an error.
                 connection = None
 
+            except Exception as err:
```

Comment:
New to the party but why are you deleting these instead of deferring them too?  Seems like a harsh thing to force on people not using SMTPlibs.

Reply:
This is the "deferring permanent errors isn't helpful, and can lead to other problems" case discussed in #73. IIRC, one of the reasons I didn't follow through with this PR is that it's not at all clear how to reliably distinguish "transient error, should defer and retry later" from "permanent error, retry won't be helpful". More discussion in #73.

If you have an idea for an approach that would work well with a variety of email backends, it'd be awesome to have someone adopt this PR and drive it to completion.

Reply:
Sorry, missed the "other problems" and that is indeed a concern.  Thanks for the update, I had read that you were stepping away from the project so I appreciate the comment.